library/src/methods/getDefaults/getDefaults.ts

```diff
@@ -22,36 +16,24 @@ import type { DefaultValues } from './types.ts';
  *
  * @returns The default values.
  */
-export function getDefaults<
-  TSchema extends SchemaWithMaybeDefault<
-    BaseSchema | ObjectSchema<ObjectEntries, any> | TupleSchema<TupleItems, any>
-  >
->(schema: TSchema): DefaultValues<TSchema> {
-  // Create defaults variable
-  let defaults: any;
-
+export function getDefaults<TSchema extends SchemaWithMaybeDefault>(
+  schema: TSchema
+): DefaultValues<TSchema> | undefined {
   // If schema contains a default function, set its default value
   if (schema.default) {
-    defaults = getDefault(schema);
-
-    // Otherwise, check if schema is of kind object or tuple
-  } else if ('type' in schema) {
-    // If it is an object schema, set object with default value of each entry
-    if (schema.type === 'object') {
-      defaults = {};
-      for (const key in schema.entries) {
-        defaults[key] = getDefaults(schema.entries[key]);
-      }
-
-      // If it is a tuple schema, set array with default value of each item
-    } else if (schema.type === 'tuple') {
-      defaults = [];
-      for (let key = 0; key < schema.items.length; key++) {
-        defaults.push(getDefaults(schema.items[key]));
-      }
-    }
+    return getDefault(schema);
+  }
+  // Otherwise, check if schema is of kind object or tuple
+  // If it is an object schema, set object with default value of each entry
+  if (isObjectSchema(schema)) {
+    return Object.entries(schema.entries).reduce(
```

Comment:
If using `Object.fromEntries` in the async version, why not use it here as well? Is it because of the performance cost of creating an extra array?

Reply:
I suppose it can be done with `Object.fromEntries` here too, since in either case a type assertion is required. I implemented one after the other and solved for type errors along the way, so these were just the solutions I arrived at then. Will make this change when I resolve merge conflicts.